src/omni_epd/displays/waveshare_display.py

```diff
@@ -77,3 +99,159 @@ def close(self):
+
+    def sleep(self):
+        self._device.sleep()
+
+    def clear(self):
+        self._device.Clear()
+
+    def close(self):
+        # can't import this earlier as pkg may not be installed
+        from waveshare_epd import epdconfig
+        epdconfig.module_init()
+        epdconfig.module_exit()
+
+
+class Waveshare102inDisplay(VirtualEPD):
+    """
+    This class is for the Waveshare 1.02 in display only as it has some method calls that are different
+    https://github.com/waveshare/e-Paper/blob/master/RaspberryPi_JetsonNano/python/lib/waveshare_epd/epd1in02.py
+    """
+
+    pkg_name = 'waveshare_epd'
+
+    def __init__(self, deviceName, config):
+        super(Waveshare102inDisplay, self).__init__(deviceName, config)
+
+        deviceObj = self.load_display_driver(self.pkg_name, deviceName)
+
+        # create the epd object
+        self._device = deviceObj.EPD()
+
+        # set the width and height
+        self.width = self._device.width
+        self.height = self._device.height
+
+    @staticmethod
+    def get_supported_devices():
+        result = []
+
+        if(Waveshare102inDisplay.check_module_installed('waveshare_epd')):
+            result = [f"{Waveshare102inDisplay.pkg_name}.epd1in02"]
+
+        return result
+
+    def prepare(self):
+        self._device.Init()
+
+    def _display(self, image):
+        self._device.Display(self._device.getbuffer(image))
+
+    def sleep(self):
+        self._device.Sleep()
+
+    def clear(self):
+        self._device.Clear()
+
+    def close(self):
+        # can't import this earlier as pkg may not be installed
+        from waveshare_epd import epdconfig
+        epdconfig.module_init()
+        epdconfig.module_exit()
+
+
+class Waveshare3in7Display(VirtualEPD):
+    """
+    This class is for the Waveshare 3.7in display only as it
+    has support for different gray scales and the methods are different
+    https://github.com/waveshare/e-Paper/blob/master/RaspberryPi_JetsonNano/python/lib/waveshare_epd/epd3in7.py
+    """
+
+    pkg_name = 'waveshare_epd'
+
+    def __init__(self, deviceName, config):
+        super(Waveshare3in7Display, self).__init__(deviceName, config)
+
+        deviceObj = self.load_display_driver(self.pkg_name, deviceName)
+
+        # create the epd object
+        self._device = deviceObj.EPD()
+
+        # set the width and height
+        self.width = self._device.width
+        self.height = self._device.height
+
+    @staticmethod
+    def get_supported_devices():
+        result = []
+
+        if(Waveshare3in7Display.check_module_installed('waveshare_epd')):
+            result = [f"{Waveshare102inDisplay.pkg_name}.epd3in7"]
+
+        return result
+
+    def prepare(self):
+        # only b/w supported right now
+        self._device.init(1)
+
+    def _display(self, image):
+        self._device.display_1Gray(self._device.getbuffer(image))
```

Comment:
The driver for this already does the [greyscale conversion](https://github.com/waveshare/e-Paper/blob/af9694c602fb24d776cd0f0b3e073a477294e3d8/RaspberryPi_JetsonNano/python/lib/waveshare_epd/epd3in7.py#L267), so it should just work.

Reply:
As I was going through the individual driver classes I made some notes for things that can be revisited. This one was an oddity because it didn't have a `display()` method at all so I wanted to catch that. 

For the others the main thing to do is provide a mechanism for the user to choose which they want since, like you said, the driver will do the conversion on the waveshare devices.